src/test/java/com/redhat/cloud/notifications/TestLifecycleManager.java

```diff
@@ -87,11 +88,23 @@ private void setupMockServer(Map<String, String> properties) {
         mockEngineServer.start();
         String mockServerUrl = "http://" + mockEngineServer.getContainerIpAddress() + ":" + mockEngineServer.getServerPort();
         properties.put("rbac/mp-rest/url", mockServerUrl);
+        properties.put("notifications-backend/mp-rest/url", mockServerUrl);
+
         mockServerClient = new MockServerClient(mockEngineServer.getContainerIpAddress(), mockEngineServer.getServerPort());
 
         String xRhIdentity = TestHelpers.encodeIdentityInfo("test","user");
         String access = TestHelpers.getFileAsString("rbac_example_full_access.json");
 
+        mockServerClient
+                .when(request()
+                        .withPath("/internal/validation/baet")
```

Comment:
I think you could make this more concrete to check the request with a good b/a/et and a bad one, which the would return a 404 code

Reply:
yes, but the thing is, there is an issue with the MockServer. I wanted to do that, but it didn't work.
I added two methods, one which returns 200 for valid depending on specific application, bundle and eventtype and one  returning 404, but unfortunately the MockServer didn't react on it.

Reply:
What did you try?

Reply:
things like this:

public void addMock() {
        this.mockServerClient
                .when(request()
                        .withPath("/internal/validation/baet")
                )
                .respond(response()
                        .withStatusCode(200)
                        .withHeader("Content-Type", "application/json")
                        .withBody(""));
    }

    public void addMock2() {
        this.mockServerClient
                .when(request()
                                .withPath("/internal/validation/baet")
                )
                .respond(response()
                        .withStatusCode(404)
                        .withHeader("Content-Type", "application/json")
                        .withBody("blabla"));
    }
    
without expecting specific parameters like bundle = my-bundle and even with it. 
I called addMock() in the one test and addMock2() in the other but it didn't overwrite the mock.

Reply:
You need some 'discriminator' e.g. one of the query params, that are sent to the notifications-backend.

```
when(request()
  . withPath(....)
  ..withQueryStringParameters(param(string("bundle")),string("rhel")))
  .respond(response().withStatusCode(200));
when(request()
  . withPath(....)
  ..withQueryStringParameters(param(string("bundle")),string("blabla")))
  .respond(response().withStatusCode(404));

```  

Check out https://www.mock-server.com/mock_server/creating_expectations.html#button_match_request_by_query_parameter_by_sub_set

Reply:
done